saleor/attribute/models/base.py

```diff
@@ -242,23 +242,27 @@ def bulk_get_or_create(self, objects_data):
         results = []
         objects_not_in_db: List[AttributeValue] = []
 
+        # prepare a list that will save order index of attribute values
+        objects_enumerated = [
+            (index, obj_data) for index, obj_data in enumerate(objects_data)
+        ]
         query = self._prepare_query_for_bulk_operation(objects_data)
 
         # iterate over all records in db and check if they match any of objects data
         for record in query.iterator():
             # iterate over all objects data and check if they match any of records in db
-            for obj in objects_data:
+            for index, obj in objects_enumerated:
```

Comment:
Instead of creating `objects_enumerated`, can't you just call here:
```python
for index, obj in enumerate(objects_data):
```
?

Reply:
additionally `objects_enumerated` is passed later to `_add_new_records`